docs/USAGE-GUIDE.md

```diff
@@ -828,9 +833,26 @@ Authentication is usually the most expensive part of the handshake. To avoid the
 
 ### Configuring the Trust Store
 
-To validate the peer’s certificate, the local “trust store” must contain a certificate that can authenticate the peer’s certificate.
-
-By default, s2n-tls will be initialized with the common trust store locations for the host operating system. To completely override those locations, call `s2n_config_wipe_trust_store()`. To add certificates to the trust store, call `s2n_config_set_verification_ca_location()` or `s2n_config_add_pem_to_trust_store()`.
+To validate the peer’s certificate, a config's local “trust store” must contain a certificate
+that can authenticate the peer’s certificate. To add certificates to the trust store, call
+`s2n_config_set_verification_ca_location()` or `s2n_config_add_pem_to_trust_store()`.
+
+`s2n_config_new()` initializes the trust store with the default system certificates, which may
+include public CAs or other unexpected certificates. If s2n-tls is verifying certificates and does
+not require these default system certificates, they should be cleared with
+`s2n_config_wipe_trust_store()` before calling `s2n_config_set_verification_ca_location()` or
+`s2n_config_add_pem_to_trust_store()`.
+
+Note that if s2n-tls is operating as a server performing client authentication, the
+`s2n_verify_host_fn` callback must be implemented to validate the identity of all received
+certificates. If the identity of the received certificates are implicit (i.e. the certificates are
+self-signed, or signed by a privately owned CA), the `s2n_verify_host_fn` callback need not perform
```

Comment:
I don't think we should restrict this statement to client auth. Like, at most we should mention it's more likely with client auth. This would be important to know for a server using self-signed certs as well, it's just more likely the default implementation of s2n_verify_host_fn would work for that server.